### PR TITLE
Add clearml logger

### PR DESCRIPTION
**Pull request by Fissium**
Enhanced the experiment tracking functionality by integrating ClearML

### CODE REVIEW

File: oml/lightning/pipelines/logging.py [hunk 1]
```diff
@@ -31,6 +34,66 @@ def prepare_tags(cfg: TCfg) -> List[str]:
     return tags
 
 
+class ClearMLLogger(IPipelineLogger):
```
Comment (AlekseySh) [inline oml/lightning/pipelines/logging.py:37]: not sure about inheritance here

I think `ClearMLLogger` should be a child of `LightningLogger`

and `ClearMLPipelineLogger` should be a child of `ClearMLLogger` and `IPipelineLogger`

Am I wrong?

**Review by AlekseySh — CHANGES_REQUESTED** [review]
Thank you for the contribution! I will take a deeper look later

File: oml/lightning/pipelines/logging.py (hunk 1 above)
Comment (Fissium) [reply oml/lightning/pipelines/logging.py:37]: You're right. I'll fix it.

Comment (Fissium) [reply oml/lightning/pipelines/logging.py]: Done.

File: oml/lightning/pipelines/logging.py [hunk 2]
```diff
@@ -31,6 +35,66 @@ def prepare_tags(cfg: TCfg) -> List[str]:
...
 
 
+class ClearMLLogger(Logger):
+    def __init__(self, **kwargs: Any):
+        try:
+            from clearml import Task
+        except ImportError as e:
+            raise ModuleNotFoundError(
+                "This contrib module requires clearml to be installed. "
+                "You may install clearml using: \n pip install clearml>=1.5.0 \n"
```
Comment (AlekseySh) [inline oml/lightning/pipelines/logging.py:45]: let's avoid hardcoding the exact version here, I'm sure we will forget to adjust it after we change it in requirements :)

File: tests/test_runs/test_pipelines/configs/train_arcface_with_categories.yaml [hunk 3]
```diff
@@ -51,17 +51,19 @@ metric_args:
...
+log_images: True
 
 metric_for_checkpointing: OVERALL/cmc/1
 
 max_epochs: 2
 valid_period: 1
 
 logger:
-  name: tensorboard
+  name: clearml
```
Comment (AlekseySh) [inline tests/test_runs/test_pipelines/configs/train_arcface_with_categories.yaml:62]: please, add the following string to the corresponding test:

```
@pytest.mark.long
@pytest.mark.needs_optional_dependency  # <---------- THIS ONE
@pytest.mark.parametrize("accelerator, devices", accelerator_devices_pairs())
def test_train_arcface_with_categories(accelerator: str, devices: int) -> None:
    run("train_arcface_with_categories.py", accelerator, devices)
```

**Review by AlekseySh — CHANGES_REQUESTED** [review]
only a few comments

File: tests/test_runs/test_pipelines/configs/train_arcface_with_categories.yaml (hunk 3 above)
Comment (Fissium) [reply tests/test_runs/test_pipelines/configs/train_arcface_with_categories.yaml:62]: Done

File: oml/lightning/pipelines/logging.py (hunk 2 above)
Comment (Fissium) [reply oml/lightning/pipelines/logging.py]: Done ;)

**Comment by Fissium** [comment]
It seems we need to add `clearml` to `LIBS_TO_IGNORE = ["torch_xla", "pytorch_grad_cam", "wandb", "neptune", "IPython"]` in `test_import.py`.

Am I right?

**Comment by AlekseySh** [comment]
@Fissium seems like this, yes

**Review by AlekseySh — APPROVED** [review]
Great job!

**Pull request merged**